src/app/clusters/resource-monitoring-server/resource-monitoring-cluster-proxy.h

```diff
@@ -0,0 +1,68 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include "resource-monitoring-cluster.h"
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ResourceMonitoring {
+
+template <ClusterId DISCRIMINATOR>
+class ResourceMonitoringClusterProxy : public ResourceMonitoringCluster
```

Comment:
Needs documentation.

Also, calling it "proxy" in a Matter context in which that word means something rather specific (and quite different) may not be ideal.